### DISCUSSION TITLE
Obtaining a SCALAR AuxVariable from surface integral (no postprocessor)

### DISCUSSION THREAD

**Discussion by nlosacco** [idaholab/moose#26735] [Q&A General]
Hi all,

I need to obtain the total flow through a surface in a `Diffusion` or `PorousFlow` analysis.
I know, for instance, I can store the flux vectors with the `save_in` option of `NeumannBC` and then obtain the integral (total flux through a boundary) with a `NodalSum` postprocessor. But I need to use that value as an AuxVariable, since I want to set some initial conditions for it. I tried with the `SumAux` AuxKernel, but it yields a `MONOMIAL` variable rather then a `SCALAR`. Is there a way to achieve this?

**Comment by GiudGiud** [comment]
You want to initialize the flux?

The flux is a byproduct of the simulation, computed from the nonlinear variables. Initializing it will simply get it overwritten next time it is computed with no effect on the simulation imo. 

You could initialize the nonlinear variables to get the desired initial flux though. 

For visualizing the flux, I would look in the tagging interface to try to save the flux to a tagged vector. Then a tagged vector aux should be able to output it to a nodal variable. I haven't used it that way before

**Reply by nlosacco** [reply]
I need the flux at a boundary as the coupling term between a diffusion and a ODE. If I simply use a postprocessor, the flux at t=0 will be zero, hence the results will be inaccurate at the beginning of the simulation, when the process occurs more quickly.
I kind of solved by copying the postprocessor to a scalar variable, which is first initialised with the flux at t=0 (known in the simple case I'm simulating) and adopting the proper `execute_on` flags.

**Reply by GiudGiud** [reply]
ok that makes sense. 
There are certain objects you can execute on INITIAL that would be able to initialize a variable.
A postprocessor could be executed on INITIAL, and the variable could have an initialization value using either an IC (for a nonlinear variable, so not the flux I think?) or an auxkernel executed on INITIAL for an auxiliary variable

**Reply by nlosacco** [reply]
Actually, while for very simple problems the initial flux is known a priori, and the relevant AuxVariables can be duly initialised, for more complex problems the initial flux cannot be easily worked out. Then, I'm thinking of a workaround.
I could obtain the initial flux as as postprocessor at the end of the first timestep, then repeat the analysis from the beginning initialising the flux AuxVariable with that initial flux. Is there a way to do this automatically? I mean, run a single timestep and then rerun the whole analysis from the beginning (hence resetting everything to the initial state) but setting one auxvariable to the value of the post processor obtained in the former analysis? Or do I need to do it manually?

**Reply by GiudGiud** [reply]
Hello

The cleanest way to do this is to use a MultiApp.
You run this initialization step in a separate simulation in a FullSolveMultiapp that is executed on INITIAL.
Unless you use command line arguments for the multiapp to turn off the MultiApp's MultiApp block, it will mean having two input files: one for initialization and one for the main simulation. But they will be run one after each other when running the main simulation
